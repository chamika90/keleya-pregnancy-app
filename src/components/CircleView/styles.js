import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    margin: 5,
  },
});

export default styles;
