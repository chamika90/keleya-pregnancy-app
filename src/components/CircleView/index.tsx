import React from 'react';
import {View} from 'react-native';

import styles from './styles';

interface Props {
  bgColor: string;
  size: number;
}

/*
 * Circle View
 */
const CircleView: React.FC<Props> = ({bgColor, size}) => {
  return (
    <View
      style={[
        styles.container,
        {
          backgroundColor: bgColor,
          width: size,
          height: size,
          borderRadius: size / 2,
        },
      ]}
    />
  );
};

export default CircleView;
