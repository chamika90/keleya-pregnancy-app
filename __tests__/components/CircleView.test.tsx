import React from 'react';
import 'react-native';
import {render} from '@testing-library/react-native';

import CircleView from '../../src/components/CircleView';

describe('CircleView', () => {
  it('should match snapshot', async () => {
    const {toJSON} = render(<CircleView bgColor={'#000000'} size={10} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
