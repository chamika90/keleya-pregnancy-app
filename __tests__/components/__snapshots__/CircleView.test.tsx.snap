// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleView should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "justifyContent": "center",
        "margin": 5,
      },
      Object {
        "backgroundColor": "#000000",
        "borderRadius": 5,
        "height": 10,
        "width": 10,
      },
    ]
  }
/>
`;
